ibm/data_source_ibm_appid_application.go

```diff
@@ -0,0 +1,106 @@
+package ibm
+
+import (
+	"context"
+	"fmt"
+	appid "github.com/IBM/appid-management-go-sdk/appidmanagementv4"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceIBMAppIDApplication() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceIBMAppIDApplicationRead,
+		Schema: map[string]*schema.Schema{
+			"tenant_id": {
+				Description: "The service `tenantId`",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"client_id": {
+				Description: "The `client_id` is a public identifier for applications",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"name": {
+				Description: "The application name",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"secret": {
+				Description: "The `secret` is a secret known only to the application and the authorization server",
+				Type:        schema.TypeString,
+				Computed:    true,
+				Sensitive:   true,
+			},
+			"oauth_server_url": {
+				Description: "Base URL for common OAuth endpoints, like `/authorization`, `/token` and `/publickeys`",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"profiles_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"discovery_endpoint": {
+				Description: "This URL returns OAuth Authorization Server Metadata",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"type": {
+				Description: "The type of application. Allowed types are `regularwebapp` and `singlepageapp`.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func dataSourceIBMAppIDApplicationRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
```

Comment:
I think this declaration is not required you can retrun nil in place of return diags...
it holds for all datasources and resources